### PR TITLE
Adds z index tokens to rails and js

### LINKED ISSUE #1084: [Suggestion] sage-react `SageToken`: Include z-index tokens within object
## Description
Include Sage z-indexes within `kajabi/sage-react`'s `SageTokens`.

```
$sage-z-indexes: (
  negative: -1,
  default: 0,
  raised: 1000,
  alert: 2000,
  underlay: 3000,
  nav: 4000,
  overlay: 5000,
  modal: 6000,
  priority: 8000,
  nuclear: 9000
);
```
https://sage-lib-sassdocs.herokuapp.com/#variable-sage-z-indexes

## Additional Information
I have a few instances from this cycle where I needed access to these values within ES6 JS.
Here’s a search result of the instances:
https://github.com/Kajabi/kajabi-products/search?l=JavaScript&q=Sage-modal+z-index&type=code

The purpose in each of those results is to ensure a third party lib isn’t overlapping on top of an active SageModal.

Seems worth reporting! However the current comments in the code also do the job.

### PR DESCRIPTION
**Pull request by philschanely**
## Description

Adds `Z_INDEXES` to the Rails and React `SageTokens` constant.

Closes #1084 

NOTE: Follow up to improve tokens documentation will be needed to better publicize this feature and others.

## Testing in `kajabi-products`
1. (**LOW**) Adds a new Z index token to Rails and React. No impact on existing work.

### CODE REVIEW

File: docs/lib/sage_rails/app/sage_components/sage_tokens.rb [hunk 1]
```diff
@@ -338,4 +338,18 @@ module SageTokens
     "body", "body-sm", "body-xs",
   ]
 
+  # Keep in sync with `packages/sage-assets/lib/stylesheets/tokens/_z_index.scss`
```
Comment (monicawheeler) [inline docs/lib/sage_rails/app/sage_components/sage_tokens.rb:341]: ❤️

**Review by pixelflips — APPROVED** [review]
LGTM! 👍🏼

File: packages/sage-react/lib/configs/tokens/z-indexes.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+// Keep in sync with `packages/sage-assets/lib/stylesheets/tokens/_z_index.scss`
```
Comment (monicawheeler) [inline packages/sage-react/lib/configs/tokens/z-indexes.js:1]: ❤️  Should we also note in the `_z_index.scss` file to update here as well as in `sage_tokens.rb`?

Comment (philschanely) [reply packages/sage-react/lib/configs/tokens/z-indexes.js:1]: Done

**Pull request merged**